### PR TITLE
Measuring tool should respect the project's distance and area units type

### PR DESCRIPTION
**Pull request by nirvn**
QField's measuring tool now respects these project settings:
image

We also now have sensible defaults (i.e. meters for distance and square meters for area)

@mbernasocchi , here we are :)

### CODE REVIEW

File: src/qml/qgismobileapp.qml [hunk 1]
```diff
@@ -2558,6 +2558,11 @@ ApplicationWindow {
             dashBoard.ensureEditableLayerSelected();
         }
 
+        var distanceString = iface.readProjectEntry("Measurement" ,"/DistanceUnits", "")
+        projectInfo.distanceUnits = distanceString !== "" ? UnitTypes.decodeDistanceUnit(distanceString) : UnitTypes.DistanceMeters
```
[the remark this answers is not in the document]
Comment (nirvn) [reply src/qml/qgismobileapp.qml:2562]: I'd rather stick to the logic defined in QgsProject::distanceUnits. And in the future, we should just make that QgsProject function Q_INVOKABLE.

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for b1f264f69b8d683a38411ace439f629f4ec51557: [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3276-e2d78ef30dd38ea8878967f0feb7829b197f2ccb-arm64-android.apk)
<details>
<summary>Other architectures</summary>

- [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3276-e2d78ef30dd38ea8878967f0feb7829b197f2ccb-arm-android.apk)
- [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3276-e2d78ef30dd38ea8878967f0feb7829b197f2ccb-x64-android.apk)
- [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3276-e2d78ef30dd38ea8878967f0feb7829b197f2ccb-x86-android.apk)
</details>

File: src/qml/qgismobileapp.qml (hunk 1 above)
[the remark this answers is not in the document]
Comment (m-kuhn) [reply src/qml/qgismobileapp.qml:2562]: Or even better a `Q_PROPERTY`?

[the remark this answers is not in the document]
Comment (m-kuhn) [reply src/qml/qgismobileapp.qml:2562]: If this is just a bandaid for until 3.28 is out I don't care too much :-)

**Review by m-kuhn — APPROVED** [review]

File: src/qml/qgismobileapp.qml (hunk 1 above)
[the remark this answers is not in the document]
Comment (nirvn) [reply src/qml/qgismobileapp.qml:2562]: @m-kuhn , ya sorry, Q_PROPERTY, and yeah correct, this is for us to enjoy meaningful measurement units until 3.28 comes :)

**Pull request merged**